sodetlib/operations/translate_tes_parameters.py

```diff
@@ -0,0 +1,569 @@
+# 3. Save updated IVAnalysis object as a numpy file.
+###############################################################################
+
+def model_logalpha(r, logp0=0., p1=7., p2=1., logp3=0.):
+    """
+    Fits Alpha' parameter. P0 has units of [uW]^-1.
+    """
+    return logp0 + p1 * (1-r) + p2 * r ** np.exp(logp3) * np.log(1-r)
+
+
+def calculate_rpcurve(iva, minrfrac=0.2, maxrfrac=0.9):
+    """
+    Fit RP curve from IVA data. minrfrac and maxrfrac define the range
+    in which the RP curve is fit over.
+    
+    Args:
+    ----
+    iva : IVAnalysis
+    
+    
+    Returns:
+    --------
+    popts : np.array of shape (ndets x 6)
+        Fitted parameters for logp0, p1, p2, logp3, P_opt, and drn
+        P0 is optical power
+    """
+    nchans = iva['nchans']
+    popt_size = 6
+    popts = np.full((nchans, popt_size), np.nan)
+    
+    r1 = np.linspace(min(minrfrac, 0.0001), max(maxrfrac, 0.9999), 9999)
+    dr = r1[1] - r1[0]
+
+    def model_Pj(r, logp0=0., p1=7., p2=1., logp3=1.,
+         P_opt=0., drn=0.):
+        loga1 = model_logalpha(r1, logp0=logp0, p1=p1, p2=p2, logp3=logp3)
+        P1 = np.cumsum(dr / (np.exp(loga1) * r1))
+        P1 -= np.interp(0.5, r1, P1)
+        return np.interp(r / (1. + drn), r1, P1) + P_opt
+
+    for ch in range(nchans):
+        try:
+            bg = iva['bgmap'][ch]
+            if bg == -1:
+                continue
+                
+            isc = iva['idxs'][ch][0]
+            R = iva['R_V1'][ch][isc:] 
+            Rn = iva['R_n_V1'][ch]
+            r = R / Rn
+            P = iva['p_tes_V1'][ch][isc:] * 1e12
+            ok = (r > minrfrac) * (r < maxrfrac)
+            p0_guess = [0,3,1,1,0,0]        
+            RP_popts, pcov = curve_fit(model_Pj, r[ok], P[ok], p0=p0_guess)
+            
+        except:
+            continue
+    
+        popts[ch,:] = RP_popts
+    
+    iva['RP_popts'] = popts
+    return iva['RP_popts']
+
+
+def recompute_psats(iva2, psat_level=0.9):
+    """
+    Re-computes Psat for an IVAnalysis object. Will save results to iva.p_sat.
+    This assumes i_tes, v_tes, and r_tes have already been calculated.
+
+    Args
+    ----
+    iva2 : Dictionary
+        Dictionary built from original IV Analysis .npy file
+    psat_level : float
+        R_frac level for which Psat is defined. If 0.9, Psat will be the
+        power on the TES when R_frac = 0.9.
+
+    Returns
+    -------
+    p_sat_V1 : np.ndarray
+        Array of length (nchan) with the p-sat computed for each channel (W)
+    """
+    # calculates P_sat as P_TES when Rfrac = psat_level
+    # if the TES is at 90% R_n more than once, just take the first crossing
+    iva2['p_sat_V1'] = np.full(iva2['p_sat'].shape, np.nan)
+    
+    for i in range(iva2['nchans']):
+        if np.isnan(iva2['R_n_V1'][i]):
+            continue
+
+        level = psat_level
+        R = iva2['R_V1'][i]
+        R_n = iva2['R_n_V1'][i]
+        p_tes = iva2['p_tes_V1'][i]
```

Comment:
For these, and all other corrected values in both the iva and bsa objects, can you replace the existing keys (of the new deepcopied dicts) instead of adding new ones?